inception-search-core/src/main/java/de/tudarmstadt/ukp/inception/search/SearchResult.java

```diff
@@ -126,4 +126,65 @@ public void setDocumentTitle(String aDocumentTitle)
     {
         documentTitle = aDocumentTitle;
     }
+
+    @Override
+    public int hashCode()
+    {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((documentTitle == null) ? 0 : documentTitle.hashCode());
+        result = prime * result + ((leftContext == null) ? 0 : leftContext.hashCode());
+        result = prime * result + offsetEnd;
+        result = prime * result + offsetStart;
+        result = prime * result + ((rightContext == null) ? 0 : rightContext.hashCode());
+        result = prime * result + ((text == null) ? 0 : text.hashCode());
+        result = prime * result + tokenLength;
+        result = prime * result + tokenStart;
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj)
```

Comment:
Please use `Objects.equals`

Reply:
Eclipse's auto-generated hashcode/equals do not make use of `Objects.equals`. It is quite annoying to manually write hashcode/equals, even with these convenience functions. 

There is an open issue for Eclipse to support `Objects`: https://bugs.eclipse.org/bugs/show_bug.cgi?id=424214

There also seems to be a recent plugin which generates the more compact versions - we could try that: https://marketplace.eclipse.org/content/jenerate

But IMHO not using that is not a blocker as long as the hashcode/equals are auto-generated from the correct set of fields.